File: app/src/main/java/com/iita/akilimo/views/activities/SplashActivity.java
package com.iita.akilimo.views.activities;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;

import com.iita.akilimo.inherit.BaseActivity;
import com.iita.akilimo.utils.FireBaseConfig;

public class SplashActivity extends BaseActivity {
    private String LOG_TAG = SplashActivity.class.getSimpleName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Thread background = new Thread() {
            public void run() {
                try {
//                    Intent intent = new Intent(SplashActivity.this, MapBoxActivity.class);
                    Intent intent = new Intent(SplashActivity.this, HomeActivity.class);
//                    Intent intent = new Intent(SplashActivity.this, FertilizersActivity.class);
//                    Intent intent = new Intent(SplashActivity.this, MarketOutletActivity.class);
                    startActivity(intent);
                    closeActivity(false);

                } catch (Exception ex) {
//                    Crashlytics.log(Log.ERROR, LOG_TAG, ex.getStepTitle());
//                    Crashlytics.logException(ex);
                    Log.e(LOG_TAG, ex.getMessage());

                }
            }
        };
        background.start();

        FireBaseConfig fireBaseConfig = new FireBaseConfig(this);
        fireBaseConfig.fetchNewRemoteConfig();
    }

    @Override
    protected void initToolbar() {
    }

    @Override
    protected void initComponent() {
    }

    @Override
    protected void closeActivity(boolean backPressed) {
        finish();
    }
}

File: app/src/main/java/com/iita/akilimo/views/activities/SplashActivity.kt
package com.iita.akilimo.views.activities

import android.content.Intent
import android.os.Bundle
import com.blogspot.atifsoftwares.animatoolib.Animatoo
import com.google.firebase.crashlytics.FirebaseCrashlytics
import com.iita.akilimo.BuildConfig
import com.iita.akilimo.dao.AppDatabase.Companion.getDatabase
import com.iita.akilimo.inherit.BaseActivity


class SplashActivity : BaseActivity() {
    val LOG_TAG: String = SplashActivity::class.java.simpleName

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initComponent()
    }

    override fun initComponent() {
        try {
            val background = object : Thread() {
                override fun run() {
                    launchActivity()
                }
            }
            background.start()
        } catch (ex: Exception) {
            FirebaseCrashlytics.getInstance().log(ex.message!!)
            FirebaseCrashlytics.getInstance().recordException(ex)
            launchActivity()
        }
    }

    override fun validate(backPressed: Boolean) {
    }

    override fun initToolbar() {
    }

    private fun launchActivity() {

        try {
            if (!BuildConfig.DEBUG) {
                getDatabase(this)?.clearAllTables()
            }
        } catch (ex: Exception) {
            FirebaseCrashlytics.getInstance().log(ex.message!!)
            FirebaseCrashlytics.getInstance().recordException(ex)
        }
        val intent = Intent(this@SplashActivity, HomeStepperActivity::class.java)
        startActivity(intent)
        finish()
        Animatoo.animateSlideDown(this)

    }
}
